src/js/view/utils/visibility.js

```diff
@@ -17,14 +17,15 @@ export default function getVisibility(model, element) {
     // Otherwise, set it to the intersection ratio reported from the intersection observer
     let intersectionRatio = model.get('intersectionRatio');
 
-    if (intersectionRatio === undefined) {
+    // Compute visibility does not return accurate value for iFrame player
+    if (intersectionRatio === undefined && !(window.top !== window.self)) {
```

Comment:
Get the value of `computeVisibility(element);`. Then if it's > 0 and we're in an iFrame, throw it away. But if it comes back as 0 in an iFrame, we can trust that the player is not visible in the iFrame and therefor impossible to view in the page.